libs/ledger/src/chain/main_chain.cpp

```diff
@@ -1544,7 +1544,7 @@ DigestSet MainChain::DetectDuplicateTransactions(BlockHash const &starting_hash,
 
   bloom_filter_false_positive_count_->add(false_positives);
 
-  if (!bloom_filter_->ReportFalsePositives(false_positives))
+  if (bloom_filter_->ReportFalsePositives(false_positives))
```

Comment:
It seems this change offsets the other one - either invert the inequality in `bloom_filter.cpp` or remove the negation here, but not both.

I'd say we keep the inequality as it was - my (un)original thinking was that returning `false` corresponds to some failure or otherwise undesirable condition, but I got the logic in this line the wrong way around.

Reply:
Both is necessary

Reply:
Fair enough. In that case please update the docstring in `bloom_filter.hpp` to reflect the new behaviour.